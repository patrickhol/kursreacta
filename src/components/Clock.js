import React from 'react'

export const Clock = ({ className, minutes = 10, seconds = 33 }) => {
    return <h2 className={`clock ${className}`}>PozostaÅ‚o {minutes}:{seconds}</h2>;
}
