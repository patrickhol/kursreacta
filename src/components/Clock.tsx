import * as React from 'react';

interface ClockInterface {
  className: string;
  minutes: number;
  seconds: number;
}

export const Clock = ({
  className,
  minutes = 10,
  seconds = 33,
}: ClockInterface) => {
  return (
    <h2 className={`clock ${className}`}>
      PozostaÅ‚o {minutes > 9 ? minutes : `0${minutes}`}:{seconds>9? seconds: `0${seconds}`}
    </h2>
  );
};
