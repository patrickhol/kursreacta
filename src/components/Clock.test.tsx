import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';

import {Clock} from './Clock';

var clockRenderer: any = null;
var root: any = null;
describe('<Clock', () => {
  describe('Given minutes and seconds (DOM)', () => {
    beforeEach(() => {
      root = document.createElement('div');
      ReactDOM.render(
        <Clock className={'inactive'} minutes={10} seconds={16} />,
        root,
      );
    });
    it('then renders H2', () => {
      expect(root.childNodes[0].nodeName).toEqual('H2');
    });
    it('then renders classNames clock inactive ', () => {
      expect(root.childNodes[0].className).toMatch(/clock inactive/);
    });
    it('then renders time 10:16 ', () => {
      expect(root.childNodes[0].textContent).toMatch(/10:16/);
    });

    it('then renders Pozostało', () => {
      expect(root.childNodes[0].textContent).toMatch(/Pozostało/);
    });
  });

  describe('Given minutes and seconds (TestRenderer)', () => {
    beforeEach(() => {
      clockRenderer = renderer.create(
        <Clock className={'inactive'} minutes={10} seconds={16} />,
      );
    });
    it('then renders H2', () => {
      expect(clockRenderer.toJSON().type).toEqual('h2');
    });
    it('then renders classNames clock inactive ', () => {
      expect(clockRenderer.toJSON().props).toMatchObject({
        className: expect.stringMatching(/clock inactive/),
      });
    });
    it('then renders time 10:16 ', () => {
      expect(clockRenderer.toJSON().children).toEqual(
        expect.arrayContaining(['10', '16']),
      );
    });
    it('then renders snapshot Clock ', () => {
      expect(clockRenderer.toJSON()).toMatchSnapshot();
    });
  });
});
