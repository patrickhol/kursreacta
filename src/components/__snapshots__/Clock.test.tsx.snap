// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Clock Given minutes and seconds (TestRenderer) then renders snapshot Clock  1`] = `
<h2
  className="clock inactive"
>
  Pozosta≈Ço 
  10
  :
  16
</h2>
`;
